pilot/pkg/serviceregistry/kube/controller/ambient/remotesecrets_test.go

```diff
@@ -0,0 +1,733 @@
+	assert.Equal(t, clusters.WaitUntilSynced(opts.Stop()), true)
+	secret0 := makeSecret(secretNamespace, "s0", clusterCredential{"c0", []byte("kubeconfig0-0")})
+	secrets := a.sec
+	t.Run("test kube config override", func(t *testing.T) {
+		secrets.Create(secret0)
+		assert.EventuallyEqual(t, func() bool {
+			return clusters.GetKey("c0") != nil
+		}, true)
+		assert.Equal(t, fakeRestConfig, &rest.Config{
+			QPS:   expectedQPS,
+			Burst: expectedBurst,
+		})
+	})
+}
+
+func TestingBuildClientsFromConfig(kubeConfig []byte, c cluster.ID, configOverrides ...func(*rest.Config)) (kube.Client, error) {
+	return kube.NewFakeClient(), nil
+}
+
+type testController struct {
+	clusters krt.Collection[*multicluster.Cluster]
+
+	client  kube.Client
+	t       *testing.T
+	secrets clienttest.TestWriter[*corev1.Secret]
+	mesh    meshwatcher.WatcherCollection
+}
+
+func buildTestController(t *testing.T) testController {
+	tc := testController{
+		client: kube.NewFakeClient(),
+		t:      t,
+	}
+	watcher := meshwatcher.NewTestWatcher(nil)
+	options := Options{
+		Client:          tc.client,
+		ClusterID:       "local-cluster",
+		SystemNamespace: "istio-system",
+		DomainSuffix:    "company.com",
+		MeshConfig:      watcher,
+		ClientBuilder:   TestingBuildClientsFromConfig,
+	}
+	t.Cleanup(options.Client.Shutdown)
+	a := newAmbientTestServerFromOptions(t, testNW, options)
+	tc.clusters = a.remoteClusters
+	tc.mesh = watcher
+	tc.secrets = a.sec
+	return tc
+}
+
+var kubeconfig = 0
+
+func (t *testController) AddSecret(secretName, clusterID string) {
+	kubeconfig++
+	t.secrets.CreateOrUpdate(makeSecret(secretNamespace, secretName, clusterCredential{clusterID, []byte(fmt.Sprintf("kubeconfig-%d", kubeconfig))}))
+}
+
+func (t *testController) DeleteSecret(secretName string) {
+	t.t.Helper()
+	t.secrets.Delete(secretName, secretNamespace)
+}
+
+func TestListRemoteClusters(t *testing.T) {
+	test.SetForTest(t, &features.EnableAmbientMultiNetwork, true)
+	tc := buildTestController(t)
+	tc.AddSecret("s0", "c0")
+	tc.AddSecret("s1", "c1")
+	// before sync
+	getSimpleClusters := func() []simpleCluster {
+		clusters := tc.clusters.List()
+		sortedClusters := slices.SortBy(clusters, func(c *multicluster.Cluster) cluster.ID {
+			return c.ID
+		})
+		return slices.Map(sortedClusters, func(c *multicluster.Cluster) simpleCluster {
+			sc := simpleCluster{
+				ID:           c.ID,
+				InitialSync:  c.HasSynced() && !c.SyncDidTimeout(),
+				SourceSecret: c.SourceSecret,
+			}
+			return sc
+		})
+	}
+
+	assert.EventuallyEqual(t, getSimpleClusters, []simpleCluster{
+		{ID: "c0", SourceSecret: types.NamespacedName{Name: "s0", Namespace: "istio-system"}, InitialSync: true},
+		{ID: "c1", SourceSecret: types.NamespacedName{Name: "s1", Namespace: "istio-system"}, InitialSync: true},
+	})
+}
+
+func TestShutdown(t *testing.T) {
+	test.SetForTest(t, &features.EnableAmbientMultiNetwork, true)
+	stop := make(chan struct{}) // Don't use test stop because we manually close it
```

Comment:
stop isn't referenced other than in `close` below, not sure I understand how it effects the test?